core/src/main/scala/scalafix/rewrite/RewriteCtx.scala

```diff
@@ -1,11 +1,29 @@
 package scalafix.rewrite
+import scala.meta.Tree
+import scala.meta.tokens.Tokens
 import scalafix.ScalafixConfig
 import scalafix.util.AssociatedComments
 import scalafix.util.TokenList
 
 case class RewriteCtx(
     config: ScalafixConfig,
+    tokens: Tokens,
     tokenList: TokenList,
     comments: AssociatedComments,
     semantic: Option[SemanticApi]
 )
+
+object RewriteCtx {
+  def fromCode(ast: Tree,
+               config: ScalafixConfig = ScalafixConfig(),
+               semanticApi: Option[SemanticApi] = None): RewriteCtx = {
```

Comment:
Good question. It depends on what rewrite is used, some rewrites like procedure syntax work 100% without the semantic api, organize imports works without it but can't handle relative imports in that case, explicit implicit will error.

I guess this could be better documented.